torch_ipex/csrc/aten_ipex_bridge.cpp

```diff
@@ -496,24 +495,52 @@ std::vector<at::Tensor> shallowFallbackToCPUTensorList(const at::TensorList& ten
   return dpcpp_tensor_vec;
 }
 
-void cvtTensorToScalaraType(const at::Tensor& ipexTensor, at::ScalarType dstScalarType) {
+
+void reorderTensorToScalarTypeForDNNL(const at::Tensor& ipexTensor, at::ScalarType dstScalarType) {
+  if (ipexTensor.device().type() == at::DeviceType::CPU) {
+    return reorderTensorToScalaraType(ipexTensor, dstScalarType);
+  }
+
+  TORCH_CHECK(dstScalarType == at::kBFloat16 || dstScalarType == at::kFloat);
+  auto tensor_dtype = ipexTensor.scalar_type();
+  TORCH_CHECK(tensor_dtype == at::kBFloat16 || tensor_dtype == at::kFloat);
+  if (tensor_dtype == dstScalarType)
+    return;
+
+  if (check_tensor_own_shade_context(ipexTensor)) {
+    // Shade data context has been attached
+    if (cpu::ShadeDataContext::isDilTensor(ipexTensor)) {
+      cpu::ShadeDataContext *shade_context = (cpu::ShadeDataContext*)(ipexTensor.storage().data_ptr().get_context());
+      shade_context->dil_tensor.to_type(get_dil_data_type(dstScalarType));
+      IPEXTensorImpl* ipex_tensor_impl = (IPEXTensorImpl *)ipexTensor.unsafeGetTensorImpl();
+      ipex_tensor_impl->reset_data_type(dstScalarType);
+      ipex_tensor_impl->storage().unsafeGetStorageImpl()->set_dtype(at::scalarTypeToTypeMeta(dstScalarType));
+      return;
+    }
+  }
+
+  return reorderTensorToScalaraType(ipexTensor, dstScalarType);
+}
+
+
+void reorderTensorToScalaraType(const at::Tensor& ipexTensor, at::ScalarType dstScalarType) {
   if (!(ipexTensor.defined()))
     return;
 
   TORCH_CHECK(dstScalarType == at::kBFloat16 || dstScalarType == at::kFloat);
-  if (ipexTensor.scalar_type() == dstScalarType)
+
+  auto tensor_dtype = ipexTensor.scalar_type();
+  TORCH_CHECK(tensor_dtype == at::kBFloat16 || tensor_dtype == at::kFloat);
+  if (tensor_dtype == dstScalarType)
     return;
 
-  if (check_data_is_part_of_storage(ipexTensor))
+  if (!check_tensor_own_whole_storage(ipexTensor))
```

Comment:
where is the gatekeeper for this branch?

Reply:
I will refine it. I intend not to support the case if the tensor data is just a part of its storage.